scripts/gitlab/check_runtime.sh

```diff
@@ -1,18 +1,39 @@
 #!/bin/sh
 #
 #
-# check for any changes in the ^runtime/ tree. if there are any changes found, 
-# it should mark the PR B2-breaksapi and "auto-fail" the PR if there 
-# isn't a change in the runtime/{kusama,polkadot}/src/lib.rs file that alters 
-# the version.
+# check for any changes in the ^runtime/ tree. if there are no changes check
+# if the substrate reference in the Cargo.lock has been changed. If so pull
+# the repo and verify if the {spec,impl}_version s have been altered since the
+# last reference. If there were changes the script will continue to check if
+# the spec_version resp impl_version of polkadot have been altered as well.
+# this will also be checked if there were changes to the runtime source files.
+#
+# If there are any changes found, it will mark the PR breaksapi and
+# "auto-fail" the PR if there isn't a change in the
+# runtime/{polkadot,kusama}/src/lib.rs file
+# that alters the version since the last release tag.
+
 
 set -e # fail on any error
 
 
-# give some context
+SUBSTRATE_REPO="https://github.com/paritytech/substrate"
+SUBSTRATE_REPO_CARGO="git\+${SUBSTRATE_REPO}\?branch=polkadot-master"
+SUBSTRATE_VERSIONS_FILE="node/runtime/src/lib.rs"
```

Comment:
Did you mean `bin/node/runtime/src/lib.rs` here?

Reply:
good catch, it must have changed when everything was rearranged in the repo.